cmd/kubeadm/app/phases/kubelet/config.go

```diff
@@ -66,7 +69,22 @@ func WriteConfigToDisk(cfg *kubeadmapi.ClusterConfiguration, kubeletDir, patches
 		}
 	}
 
-	return writeConfigBytesToDisk(kubeletBytes, kubeletDir)
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		kubeletBytes, err = applyKubeletConfigPatchFromFile(kubeletBytes, filepath.Join(kubeletDir, kubeadmconstants.KubeletInstanceConfigurationFileName), output)
+		if err != nil {
+			return errors.Wrap(err, "could not apply patches to the KubeletConfiguration")
+		}
+	}
+
+	return writeConfigBytesToDisk(kubeletBytes, kubeletDir, kubeadmconstants.KubeletConfigurationFileName)
+}
+
+// WriteInstanceConfigToDisk writes the container runtime endpoint configuration
+// to the instance configuration file in the specified kubelet directory.
+func WriteInstanceConfigToDisk(cRISocket string, kubeletDir string) error {
```

Comment:
ok, that's a tricky problem as there is no way to make the marshal ignore zero value fields.

in that case please make the function still accepted `kubeletDir string, cfg *KubeletConfiguration`, but inside the function extract the containerRuntimeEndpoint field and write it as YAML the way you do right now.

this will allows use to use the same function in more places in the future and not having it accept many different arguments for fields we care about.

Reply:
yes, i meant just creating a new KubeletConfiguration object and setting the field on it.
this is just making the function better organized instead of passing the many flags to it.

Reply:
all right.